sharding-core/sharding-core-rewrite/src/main/java/org/apache/shardingsphere/core/rewrite/sql/SQLBuilder.java

```diff
@@ -82,11 +84,21 @@ private String getConjunctionText(final SQLToken sqlToken) {
     
     private int getStartIndex(final SQLToken sqlToken) {
         int startIndex = sqlToken instanceof Substitutable ? ((Substitutable) sqlToken).getStopIndex() + 1 : sqlToken.getStartIndex();
-        return startIndex > logicSQL.length() ? logicSQL.length() : startIndex;
+        return Math.min(startIndex, logicSQL.length());
     }
     
     private int getStopIndex(final SQLToken sqlToken) {
         int currentSQLTokenIndex = sqlTokens.indexOf(sqlToken);
-        return sqlTokens.size() - 1 == currentSQLTokenIndex ? logicSQL.length() : sqlTokens.get(currentSQLTokenIndex + 1).getStartIndex();
+        if (sqlTokens.size() - 1 == currentSQLTokenIndex) {
```

Comment:
Hi, sorry for my ambiguous expression. 

Actually, i mean we can re-define the startIndex of `EncryptPredicateRightValueToken`, from my unstanding somehow we mistake the startIndex of `EncryptPredicateRightValueToken` to inculde `(`. 

Besides, the key point is that we want the result of `toStringForIn()` is complete and esay-understanding. So there is no defference whether the the result of `toStringForIn()`  is `'a', 'b'` or '('a', 'b')', but it is lexical-meaning.

Why do not we look at `EncryptPredicateRightValueTokenGenerator` which generates `EncryptPredicateRightValueToken`.

I'd like to listen to your idea. Thx

Reply:
@tristaZero I would like to try to modify `EncryptPredicateRightValueToken` again.